lib/jekyll-relative-links/generator.rb

```diff
@@ -65,7 +65,9 @@ def url_for_path(path)
       extension = File.extname(path)
       return unless markdown_extension?(extension)
 
-      page = site.pages.find { |p| p.path == path }
+      page = site.pages.find { |p| p.path == path } ||
+        site.docs_to_write.find { |p| p.relative_path == path } ||
+        site.static_files.find { |p| p.relative_path == path }
```

Comment:
I'd recommend moving this to a helper method, e.g., `find_target(path)` which iterates through pages, docs, and static_files once, rather than three times.